modules/core/src/test/java/org/apache/ignite/internal/processors/diagnostic/DiagnosticProcessorTest.java

```diff
@@ -167,24 +192,237 @@ public void testWalDirs() throws Exception {
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/cp} directory exists and contains checkpoint history snapshot and markers.
+     */
+    private static void validateCpDir(File baseDumpDir) {
+        File cpPath = new File(baseDumpDir, "cp");
+        assertTrue(cpPath.exists());
+        assertTrue(cpPath.isDirectory());
+
+        assertThat(cpPath.list().length, greaterThanOrEqualTo(3));
+        assertTrue(new File(cpPath, CheckpointMarkersStorage.EARLIEST_CP_SNAPSHOT_FILE).exists());
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/metastorage} directory exists and contains meta storage partition files.
+     */
+    private static void validateMetaStorageDir(File baseDumpDir) {
+        File metaStoragePath = new File(baseDumpDir, "metastorage");
+        assertTrue(metaStoragePath.exists());
+        assertTrue(metaStoragePath.isDirectory());
+
+        assertEquals(1, metaStoragePath.list().length);
+        // Somehow we use partition 0 instead of partition 1 for meta storage in this test, I don't know why.
+        String msPartFileName = String.format(PART_FILE_TEMPLATE, OLD_METASTORE_PARTITION);
+        assertTrue(new File(metaStoragePath, msPartFileName).exists());
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/sys-cache} directory exists and contains all necessary cache files.
+     */
+    private static void validateUtilityCacheDir(File baseDumpDir) {
+        File utilityCachePath = new File(baseDumpDir, "sys-cache");
+        assertTrue(utilityCachePath.exists());
+        assertTrue(utilityCachePath.isDirectory());
+
+        // Also includes a single partition file.
+        assertEquals(3, utilityCachePath.list().length);
+
+        // cache_data.dat
+        assertTrue(new File(utilityCachePath, CACHE_DATA_FILENAME).exists());
+
+        // index.bin
+        assertTrue(new File(utilityCachePath, INDEX_FILE_NAME).exists());
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/log} directory exists and contains log files.
+     */
+    private void validateLogsCacheDir(File baseDumpDir) {
+        File logPath = new File(baseDumpDir, "log");
+        assertTrue(logPath.exists());
+        assertTrue(logPath.isDirectory());
+
+        assertThat(logPath.list().length, greaterThanOrEqualTo(1));
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/jks} directory exists and contains all necessary files.
+     */
+    private void validateJksDir(File baseDumpDir) {
+        File jksPath = new File(baseDumpDir, "jks");
+        assertTrue(jksPath.exists());
+        assertTrue(jksPath.isDirectory());
+
+        assertEquals(2, jksPath.list().length);
+        assertTrue(new File(jksPath, "keystore.jks").exists());
+        assertTrue(new File(jksPath, "extras.txt").exists());
+    }
+
+    /** */
+    private void doTestOutputDiagnosticCorruptedPagesInfo(
+        Consumer<IgniteConfiguration> cfgClosure,
+        BiConsumer<IgniteEx, T2<Integer, Long>> validator
```

Comment:
It's a `TriConsumer` :) How about introducing a proper interface?

Reply:
I'm being too lazy, should I really do that?

Reply:
Currently it's not too easy to realize what `T2<Integer, Long>` means and what components it has

Reply:
Ok